types/block.go

```diff
@@ -296,49 +296,38 @@ type Commit struct {
 	// NOTE: The Precommits are in order of address to preserve the bonded ValidatorSet order.
 	// Any peer with a block can gossip precommits by index with a peer without recalculating the
 	// active ValidatorSet.
-	BlockID    BlockID `json:"block_id"`
-	Precommits []*Vote `json:"precommits"`
+	BlockID    BlockID      `json:"block_id"`
+	Precommits []*CommitSig `json:"precommits"`
+	RoundNum   int
```

Comment:
Why the `Num` suffix?

Reply:
if this is because of the `Height/Round()` methods, can't we just get rid of them?

Reply:
It is because of the getter functions. They are needed for`Commit` to implement `VoteSetReader`, which allows the consensus reactor to pull votes from `LastCommit` for gossiping to peers.
https://github.com/tendermint/tendermint/blob/b202edca9b42e5b457556d418300f19c0b0d0608/consensus/reactor.go#L617-L624